ethereum/blockwatch/block_watcher.go

```diff
@@ -426,6 +420,12 @@ func (w *Watcher) getLogsInBlockRange(from, to int) ([]types.Log, int) {
 			go func(index int, b *blockRange) {
 				defer wg.Done()
 
+				select {
+				case <-ctx.Done():
+					return
```

Comment:
We need to add an entry to `indexToLogResult` before returning otherwise the code below will panic. It expects there to be an entry for every index.

Let's set it to:
```
indexToLogResult[index] = logRequestResult{
					From: b.FromBlock,
					To:   b.ToBlock,
					Err:  errors.New("Parent context was canceled"),
					Logs: []types.Log{},
				}
```

Reply:
Just added this.